lib/helpers/throttle.js

```diff
@@ -1,43 +1,106 @@
 /**
- * Throttle decorator
- * @param {Function} fn
- * @param {Number} freq
- * @return {Function}
+ * Throttle decorator - limits how often a function can be called
+ * Ensures the function is called at most once per time interval
+ * 
+ * @param {Function} fn - The function to throttle
+ * @param {Number} freq - Frequency in calls per second (e.g., 10 = max 10 calls/sec)
+ * @return {Function} Returns [throttled function, flush function]
  */
 function throttle(fn, freq) {
```

Comment:
[CRITICAL] The function does not validate the freq parameter.

**Code Smell** detected

Suggested change:

```suggestion
if (freq <= 0) {
  throw new Error('Frequency must be a positive number');
}
```